src/config.js

```diff
@@ -13,41 +13,82 @@ const log = createLogger(__filename);
 type ApplyConfigToArgvParams = {|
   argv: Object,
   configObject: Object,
-  defaultValues: Object,
+  defaultValues?: Object,
+  subCommandDefaultValues: Object,
   configFileName: string,
+  commandExecuted?: string,
 |};
 
 export function applyConfigToArgv({
   argv,
   configObject,
   defaultValues = {},
+  subCommandDefaultValues,
   configFileName,
+  commandExecuted,
 }: ApplyConfigToArgvParams): Object {
-  const newArgv = {...argv};
+  let newArgv = {...argv};
   for (const option in configObject) {
-    // we assume the value was set on the CLI if the default value is
-    // not the same as that on the argv object as there is a very rare chance
-    // of this happening
-    if (camelCase(option) !== option) {
-      throw new UsageError(`The config option "${option}" must be ` +
-        `specified in camel case: "${camelCase(option)}"`);
-    }
-    const wasValueSetOnCLI = typeof(argv[option]) !== 'undefined' &&
-      (argv[option] !== defaultValues[option]);
-    if (wasValueSetOnCLI) {
-      log.debug(`Favoring CLI: ${option}=${argv[option]} over ` +
-        `configuration: ${option}=${configObject[option]}`);
+
+    validateOption(option);
+    if (option === commandExecuted) {
+      configObject = configObject[option];
+      newArgv = applyConfigToArgv({
+        argv,
+        configObject,
```

Comment:
you shouldn't need to redefine this variable since it's not used anywhere besides here as a param. Just keep `configObject` defined as a constant and adjust the param to the following:
````js
newArgv = applyConfigToArgv({
  configObject: configObject[option],
  ...
});
````